### DISCUSSION TITLE
Expression Pedal/Foot Controller assignment

### DISCUSSION THREAD

**Discussion by GodofPumpkins** [probonopd/MiniDexed#347] [Q&A]
I use a novation impulse 49 which sends expression pedal data on CC#11. Unfortunately there's no way to change what CC this is sent on in the midi keyboard itself. I see in dexed that foot control is on CC#04. Is there a simple way to change this? Or even a way to have the master volume controlled by the foot pedal? (I'd mostly be using the expression pedal for volume control). I can see where it's defined in mididevice.cpp but don't know how to change it on my raspberry pi or how to compile the files.

**Comment by arsamus** [comment]
Hi @GodofPumpkins what is exactly expression controller do? Change volume  from 0 to max or just a little? I've found this definition but I don't know the proper percentage:
>Expression is a percentage of volume (CC7).

https://anotherproducer.com/online-tools-for-musicians/midi-cc-list/

**Reply by GodofPumpkins** [reply]
Sorry, I wasn't very clear. The expression pedal plugs into my midi keyboard with a TRS jack (it's an On-Stage KEP100), the midi keyboard (novation impulse 49) reads the resistance of the pedal, and then the midi keyboard converts that into a midi CC message on CC#11. I am looking for a way for dexed to interpret this midi message as foot pedal (usually on CC#04) or maybe aftertouch, so I can program patches for dexed that I can change with the foot pedal.

**Comment by pik33** [comment]
I have a question then: does your MiniDexed recognize the Impulse at all? I have a Novation Impulse 61 and MiniDexed cannot recognize it (displaying "invalid configuration descriptor", "function not supported" and a bunch of hex dump after connected) on RPi 400

**Reply by GodofPumpkins** [reply]
The USB port of my Impulse is broken (been meaning to fix it) so I've built the classic midi input circuit to use.
